### PR TITLE
Object panel (image & shape)

### PR DESCRIPTION
**Pull request by EstherFu**
[H&I](https://docs.google.com/spreadsheets/d/1am_QZ2SkagYtC5rdtow3y5aC9ZnezR29y5n8rXJ-62Y/edit?pli=1#gid=0)

- Object panel: Image, Rect & Polygon (option panel & action panel)
- Update main Toolbar icons
- Temporary solution for long label: use short version in mobile
- Add unit and decimal control in ObjectPanelItem.Number

### CODE REVIEW

File: src/web/app/views/beambox/Right-Panels/Options-Blocks/PolygonOptions.spec.tsx [hunk 1]
```diff
@@ -19,47 +26,104 @@ jest.mock('helpers/i18n', () => ({
...
+    expect(getByText('.').parentElement).toHaveClass('adm-button-disabled');
+    expect(window.polygonAddSides).not.toHaveBeenCalled();
+    expect(window.polygonDecreaseSides).not.toHaveBeenCalled();
+
+    fireEvent.click(baseElement.querySelectorAll('.input-keys button')[11]);
+    expect(window.polygonAddSides).not.toHaveBeenCalled();
+    expect(window.polygonDecreaseSides).toHaveBeenCalledTimes(5);
+    expect(expect(displayBtn).toHaveTextContent('3'));
+  });
+})
```
Comment (dkdkwizard) [inline src/web/app/views/beambox/Right-Panels/Options-Blocks/PolygonOptions.spec.tsx:129]: Please end an empty line at the end of file

File: src/web/app/views/beambox/Right-Panels/Options-Blocks/RectOptions.spec.tsx [hunk 2]
```diff
@@ -74,3 +81,47 @@ describe('should render correctly', () => {
...
+    const { container } = render(
+      <RectOptions
+        elem={document.getElementById('flux')}
+        rx={10}
+        updateDimensionValues={jest.fn()}
+      />
+    );
+    expect(container).toMatchSnapshot();
+  });
+});
```
Comment (dkdkwizard) [inline src/web/app/views/beambox/Right-Panels/Options-Blocks/RectOptions.spec.tsx:127]: Please end an empty line at the end of file

File: src/web/app/views/beambox/Right-Panels/ActionsPanel.tsx [hunk 3]
```diff
@@ -212,16 +218,43 @@ class ActionsPanel extends React.Component<Props> {
...
-      this.renderButtons(
+      remove: this.renderButtons(
         'Potrace',
         () => imageEdit.potrace(elem as SVGImageElement),
         false,
         'remove',
         false,
-        <ActionPanelIcons.Invert className={styles.icon} />
+        <TraceIcon />,
+        <ObjectPanelIcons.Trace />
```
Comment (dkdkwizard) [inline src/web/app/views/beambox/Right-Panels/ActionsPanel.tsx:228]: new trace and potrace icon had been added in the develop branch, please check https://github.com/flux3dp/beam-studio-core/commit/62ac9d04d1be163598253c2cd39e2352a65a5a0d

File: src/web/app/components/mobile/CanvasTabBar.tsx [hunk 4]
```diff
@@ -45,43 +44,43 @@ const CanvasTabBar = (): JSX.Element => {
     {
       key: 'camera',
       title: lang.beambox.left_panel.label.preview,
-      icon: <CameraIcon />
+      icon: <CameraIcon style={{}} />,
```
[the remark this answers is not in the document]
Comment (EstherFu) [reply src/web/app/components/mobile/CanvasTabBar.tsx:47]: To overwrite the default style `{ marginRight: '0.2em' }` in `icons.tsx`

**Review by dkdkwizard — APPROVED** [review]

**Pull request merged**